java/code/src/com/redhat/rhn/frontend/servlets/SystemDetailsMessageFilter.java

```diff
@@ -57,10 +57,20 @@ public void doFilter(
         RequestContext rctx = new RequestContext(req);
         Long sid = rctx.getRequiredParam("sid");
         User user = rctx.getCurrentUser();
-        Server s  = SystemManager.lookupByIdAndUser(sid, user);
-        s.asMinionServer().ifPresentOrElse(
+        Server s = SystemManager.lookupByIdAndUser(sid, user);
+        this.processSystemMessages(req, s, sid);
+    }
+
+    /**
+     * Process a system and add the necessary warning messages for it if necessary
+     * @param req - the servlet request
+     * @param server - the system to be processed
+     * @param sid - the id of the system
+     */
+    public void processSystemMessages(HttpServletRequest req, Server server, Long sid) {
+        server.asMinionServer().ifPresentOrElse(
             minion -> processMinionMessages(req, minion),
-            () -> addMessageIfNecessary(req, true, TRADITIONAL_STACK_MESSAGE_KEY)
+            () -> addMessageIfNecessary(req, SystemManager.isTraditional(sid), TRADITIONAL_STACK_MESSAGE_KEY)
```

Comment:
We already have a function to check the entitlements of the system. I think you can use:
```
() -> addMessageIfNecessary(req, SystemManager.hasEntitlement(sid, EntitlementManager.MANAGEMENT), TRADITIONAL_STACK_MESSAGE_KEY)
```
instead